src/environmentd/src/lib.rs

```diff
@@ -296,10 +300,82 @@ pub async fn serve(config: Config) -> Result<Server, anyhow::Error> {
             tracing_handle: config.tracing_handle,
             adapter_client_rx: internal_http_adapter_client_rx,
             active_connection_count: Arc::clone(&active_connection_count),
+            promote_leader: promote_leader_tx,
+            ready_to_promote: ready_to_promote_rx,
         });
         server::serve(internal_http_conns, internal_http_server)
     });
 
+    'leader_promotion: {
```

Comment:
If there's this many break points, I'd encourage making this a normal function instead so `?` or `return` can be used.

Reply:
+1 it's a bit tough to follow the control flow here. FWIW named blocks are quite new so I'm not sure there are any best practices with them yet

Reply:
i just put up that diff in 7368d9bd0cd5ce1acb9e02bde36c2c75cbfd6a22. i prefer the old version: we keep the linear control flow and don't need to pass a bunch of arguments (i can't pass in `config` b/c a few lines above we partially move data out of it). The `break 'leader_promotion` become `return Ok(())` which doesn't seem as big of an improvement. What do you all think?

Reply:
Kept the diff in the list of commits for the PR but reverted it